qip-0006.md

```diff
@@ -0,0 +1,512 @@
+necessary. This includes adding a commitment to the SDF Challenge and removing
+three fields: parentHash[2], Number[2], and Nonce.
+
+#### Modified Header Fields ####
+
+| Field            | Type/Value       |
+|------------------|------------------|
+| ParentHash[0]    | common.Hash      |
+| ParentHash[1]    | common.Hash      |
+| UncleHash        | common.Hash      |
+| Coinbase         | common.Address   |
+| Root             | common.Hash      |
+| TxHash           | common.Hash      |
+| EtxHash          | common.Hash      |
+| EtxRollupHash    | common.Hash      |
+| ManifestHash[0]  | common.Hash      |
+| ManifestHash[1]  | common.Hash      |
+| ManifestHash[2]  | common.Hash      |
+| ReceiptHash      | common.Hash      |
+| Number[0]        | *big.Int         |
+| Number[1]        | *big.Int         |
+| GasLimit         | uint64           |
+| GasUsed          | uint64           |
+| BaseFee          | *big.Int         |
+| Difficulty       | *big.Int         |
+| Location         | []byte           |
+| Time             | uint64           |
+| Extra            | []byte           |
+| SDFChallenge     | common.Hash      |
+
+### SDF Challenge ###
+
+The initial SDF challenge hash will be created by taking the
+$Hash(parentHash[2],root)$. A miner can choose to sample the SDF challenge at
+any increment of $\kappa/10$ iterations. Once the $Hash_{SDF}$ is sampled it
+will be fed into a deterministic function, nonce range, which will create a
+valid nonce number range of $\kappa*10$. The miner will be able to then mine
+through the nonce range provided to search for a valid header. A new validation
+step will be added to the header in which the miner will recalculate its reward
+based on the current $n$ rounds achieved in the SDF challenge using the SDF
+challenge reward function. Additionally, a header will only have a valid nonce
+if the nonce is within the nonce range specified by the included SDF challenge
+hash.
+
+This will make miners dependent on the SDF challenge to mine valid blocks, as
+well as incentivize miners to find an SDF challenge which has the greatest
+number of rounds.  The conversion between delay and iterations $\kappa$ will
+then be determined by summing the total rounds over the previous 100 blocks and
+dividing by the difference in seconds between the block time of the first and
+last block.
+
+## Bitshare ##
+
+The interesting implication of MMTX is the creation of a new datastructure that
+will be referred to as a bitshare.  All data that may be inserted or appended
+to the blockchain highest abstract form is now a bitshare.  More specifically,
+a MMTX must take the form of a bitshare.  Depending on the weight of the
+bitshare, it will become either a header or a MMTX.
+
+### Bitshare Structure ###
+
+| Field       | Type          |
+|-------------|---------------|
+| HeaderHash  | common.Hash   |
+| ParentHash[2]  | common.Hash   |
+| Number[2] | *big.Int |
```

Comment:
Since zone parent and number have moved into the WorkObject, the block header should now only contain the dom parent hashes and dom numbers.

Reply:
also, the way its written now, it looks like there's an array of 2 ParentHashes and 2 Numbers in the Bitshare